src/content/controller_content.ts

```diff
@@ -110,10 +110,20 @@ function* ParserController() {
             while (true) {
                 const keyevent: KeyEventLike = yield
 
+                const shadowRoot =
+                    keyevent instanceof KeyboardEvent
+                        ? (keyevent.target as Element).shadowRoot
+                        : null
+
                 // _just to be safe_, cache this to make the following
                 // code more thread-safe.
                 const currentMode = contentState.mode
-                const textEditable = keyevent instanceof KeyboardEvent ? isTextEditable(keyevent.target as Element) : false
+                const textEditable =
+                    keyevent instanceof KeyboardEvent
+                        ? shadowRoot === null
+                            ? isTextEditable(keyevent.target as Element)
+                            : isTextEditable(shadowRoot.activeElement)
```

Comment:
Does any of this code need to be specific to keyboard events?

I ask because "isTextEditable" appears in a few places and `gi` doesn't work on inputs in the shadow DOM. It would be nice if it just worked everywhere.

```
src/content.ts
293:            dom.isTextEditable(document.activeElement) &&
302:            !dom.isTextEditable(document.activeElement)

src/lib/dom.ts
23:export function isTextEditable(element: Element) {
513:    if (isTextEditable(elem)) {
544:        if (isTextEditable(e.target as HTMLElement)) {

src/content/controller_content.ts
1:import { isTextEditable } from "@src/lib/dom"
116:                const textEditable = keyevent instanceof KeyboardEvent ? isTextEditable(keyevent.target as Element) : false
```